build-system/pr-check/checks.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script performs quick checks on the source code
+ * prior to running unit and integration tests.
+ * This is run during the CI stage = build; job = checks.
+ */
+
+const {
+  printChangeSummary,
+  startTimer,
+  stopTimer,
+  timedExecOrDie: timedExecOrDieBase} = require('./utils');
+const {determineBuildTargets} = require('./build-target');
+const {isTravisPullRequestBuild} = require('../travis');
+
+const FILENAME = 'checks.js';
+const timedExecOrDie =
+  (cmd, unusedFileName) => timedExecOrDieBase(cmd, FILENAME);
+
+function main() {
+  const startTime = startTimer(FILENAME, FILENAME);
+  const buildTargets = determineBuildTargets();
+  printChangeSummary(FILENAME);
+
+  timedExecOrDie('gulp update-packages');
+  timedExecOrDie('gulp presubmit');
+  timedExecOrDie('gulp lint');
+
+  if (!isTravisPullRequestBuild()) {
+    timedExecOrDie('gulp ava');
+    timedExecOrDie('node node_modules/jest/bin/jest.js');
+    timedExecOrDie('gulp check-types');
```

Comment:
`check-types` has the longest running time, so move it to the bottom after `dep-check` so this job will fail faster on average.